joblib/parallel.py

```diff
@@ -678,7 +675,7 @@ def retrieve(self):
                     self._output.extend(job.get(timeout=self.timeout))
                 else:
                     self._output.extend(job.get())
-            except tuple(self.exceptions) as exception:
+            except Exception as exception:
```

Comment:
> Also KeyboardInterrupt doesn't derive from Exception:

> In [2]: KeyboardInterrupt.__mro__
> Out[2]: (KeyboardInterrupt, BaseException, object)

Maybe there should be a comment pointing this out here, to avoid people
doing such a change in the long run.